pkg/serializer/frame/utils.go

```diff
@@ -0,0 +1,96 @@
+package frame
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing/iotest"
+)
+
+// FrameList is a list of frames (byte arrays), used for convenience functions
+type FrameList [][]byte
+
+// ReadFrameList is a convenience method that reads all available frames from the Reader
+// into a returned FrameList. If an error occurs, reading stops and the error is returned.
+func ReadFrameList(ctx context.Context, fr Reader) (FrameList, error) {
+	var frameList FrameList
+	for {
+		// Read until we get io.EOF or an error
+		frame, err := fr.ReadFrame(ctx)
+		if errors.Is(err, io.EOF) {
+			break
+		} else if err != nil {
+			return nil, err
+		}
+		// Append all frames to the returned list
+		frameList = append(frameList, frame)
+	}
+	return frameList, nil
+}
+
+// WriteFrameList is a convenience method that writes a set of frames to a Writer.
+// If an error occurs, writing stops and the error is returned.
+func WriteFrameList(ctx context.Context, fw Writer, frameList FrameList) error {
+	// Loop all frames in the list, and write them individually to the Writer
+	for _, frame := range frameList {
```

Comment:
I'm wondering what we could do to simplify this code so that it didn't need a comment?